File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/injectfragment/InjectFragmentActivityInterceptor.java
package com.raxdenstudios.square.interceptor.commons.injectfragment;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.view.View;

import com.raxdenstudios.square.interceptor.ActivityInterceptor;
import com.raxdenstudios.square.utils.FragmentUtils;

/**
 * Created by Ángel Gómez on 20/12/2016.
 */
public class InjectFragmentActivityInterceptor<TFragment extends Fragment> extends ActivityInterceptor<InjectFragmentInterceptorCallback<TFragment>> implements InjectFragmentInterceptor {

    private TFragment mContentFragment;

    public InjectFragmentActivityInterceptor(@NonNull FragmentActivity activity) {
        super(activity);
    }

    public InjectFragmentActivityInterceptor(@NonNull FragmentActivity activity, @NonNull InjectFragmentInterceptorCallback<TFragment> callback) {
        super(activity, callback);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (mContentFragment != null)
            mContentFragment.onActivityResult(requestCode, resultCode, data);
    }

    @Override
    @SuppressWarnings("unchecked")
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        View contentView = getCallback().onLoadFragmentContainer(savedInstanceState);
        if (contentView != null) {
            if (savedInstanceState == null) {
                mContentFragment = getCallback().onCreateFragment();
                FragmentUtils.INSTANCE.loadFragment(getActivity().getSupportFragmentManager(), contentView.getId(), mContentFragment);
            } else {
                mContentFragment = (TFragment) FragmentUtils.INSTANCE.getFragment(getActivity().getSupportFragmentManager(), contentView.getId());
            }
            getCallback().onFragmentLoaded(mContentFragment);
        }
    }


}

File: square-commons/src/main/java/com/raxdenstudios/square/interceptor/commons/injectfragment/InjectFragmentActivityInterceptor.kt
package com.raxdenstudios.square.interceptor.commons.injectfragment

import android.app.Activity
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentActivity
import android.view.View
import com.raxdenstudios.square.interceptor.ActivityInterceptor

/**
 * Created by Ángel Gómez on 22/05/2015.
 */
class InjectFragmentActivityInterceptor<TFragment : Fragment>(
        callback: HasInjectFragmentInterceptor<TFragment>
) : ActivityInterceptor<InjectFragmentInterceptor, HasInjectFragmentInterceptor<TFragment>>(callback),
        InjectFragmentInterceptor {

    private lateinit var mContainerView: View
    private var mFragment: TFragment? = null

    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
        super.onActivityCreated(activity, savedInstanceState)

        getFragmentManager(activity)?.also { fm ->
            mContainerView = mCallback.onLoadFragmentContainer()
            if (savedInstanceState == null) {
                mFragment = mCallback.onCreateFragment().also {
                    fm.beginTransaction()
                            .replace(mContainerView.id, it, it.javaClass.simpleName)
                            .commit()
                    mCallback.onFragmentLoaded(it)
                }
            }
        }
    }

    override fun onActivityStarted(activity: Activity, savedInstanceState: Bundle?) {
        super.onActivityStarted(activity, savedInstanceState)

        getFragmentManager(activity)?.also { fm ->
            if (savedInstanceState != null) {
                mFragment = (fm.findFragmentById(mContainerView.id) as? TFragment)?.also {
                    mCallback.onFragmentLoaded(it)
                }
            }
        }
    }

    override fun onActivityDestroyed(activity: Activity) {
        super.onActivityDestroyed(activity)
        mFragment = null
    }
}

